std/algorithm/iteration.d

```diff
@@ -1524,14 +1540,26 @@ private struct GroupByImpl(alias pred, Range)
     }
 
     @property bool empty() { return impl.current.empty; }
-    @property auto front() { return Group(impl); }
+
+    @property auto front()
+    {
+        static if (isUnary)
+        {
+            import std.typecons : tuple;
+            return tuple(unaryFun!pred(impl.current.front), Group(impl));
```

Comment:
This kind of misses the point. The extra evaluation of the predicate should have been avoided. Otherwise, there really is no gain here.